core/src/main/java/org/apache/brooklyn/util/core/file/ArchiveUtils.java

```diff
@@ -213,8 +213,10 @@ public static void deploy(String archiveUrl, SshMachineLocation machine, String
      * @see #deploy(Map, String, SshMachineLocation, String, String, String)
      */
     public static void deploy(Map<String, ?> props, String archiveUrl, SshMachineLocation machine, String destDir) {
+        List<File> filesToDelete = MutableList.of();
```

Comment:
I  can't see why this is is a list of files if it only could content one zip file.
This will a bit faster and consume less memory
```
        File toDelete = null;
        if (Urls.isDirectory(archiveUrl)) {
            File zipFile = ArchiveBuilder.zip().entry(".", Urls.toFile(archiveUrl)).create();
            toDelete = zipFile;
            archiveUrl = zipFile.getAbsolutePath();
        }
        ...
        if(toDelete != null) toDelete.delete();
```

Reply:
it's the same pattern used elsewhere which i think is nice for consistency, though minor either way imo

Reply:
Makes sense. Thanks for the clarification, I though I was missing something.